src/index.js

```diff
@@ -0,0 +1,223 @@
+const state = {
+  currentTemp: 0,
+  cityName: "Miami",
+  defaultCityName: "Seattle",
+};
+
+/************************/
+/******* Wave 2 *********/
+/************************/
+
+// Function to update the temperature display and apply color changes
+const updateTemperatureDisplay = () => {
+  const tempValueElement = document.getElementById("tempValue");
+  const landscapeElement = document.getElementById("landscape");
+  tempValueElement.textContent = `${state.currentTemp}°`;
+
+  if (state.currentTemp >= 80) {
+    tempValueElement.style.color = "red";
+    landscapeElement.textContent = "🌵  🐍 🦂 🌵🌵  🐍 🏜 🦂";
+  } else if (state.currentTemp >= 70) {
+    tempValueElement.style.color = "orange";
+    landscapeElement.textContent = "🌸🌿🌼  🌷🌻🌿 ☘️🌱 🌻🌷";
+  } else if (state.currentTemp >= 60) {
+    tempValueElement.style.color = "salmon";
+    landscapeElement.textContent = "🌾🌾 🍃 🪨  🛤 🌾🌾🌾 🍃";
+  } else if (state.currentTemp >= 50) {
+    tempValueElement.style.color = "green";
+    landscapeElement.textContent = "🌲🌲⛄️🌲⛄️🍂🌲🍁🌲🌲⛄️🍂🌲";
+  } else if (state.currentTemp <= 49) {
+    tempValueElement.style.color = "blue";
+    landscapeElement.textContent = "⛄️ ⛄️ ⛄️";
+  }
+};
+
+// Function to increase temperature
+const increaseTemp = () => {
+  state.currentTemp += 1;
+  updateTemperatureDisplay();
+};
+
+// Function to decrease temperature
+const decreaseTemp = () => {
+  state.currentTemp -= 1;
+  updateTemperatureDisplay();
+};
+
+/** Wave 2 events **/
+const temperatureControls = () => {
+  const warmUpButton = document.getElementById("increaseTempControl");
+  const coolDownButton = document.getElementById("decreaseTempControl");
+
+  warmUpButton.addEventListener("click", increaseTemp);
+  coolDownButton.addEventListener("click", decreaseTemp);
+};
+
+
+/************************/
+/******* Wave 3 *********/
+/************************/
+
+const updateCity = () => {
+  const cityNameInput = document.getElementById("cityNameInput");
+  const currentCityHeader = document.getElementById("headerCityName");
+  currentCityHeader.textContent = cityNameInput.value;
+};
+
+const handleUpdateCityEvent = () => {
+  const cityNameInput = document.getElementById("cityNameInput");
+  const resetButton = document.getElementById("cityNameReset");
+  cityNameInput.addEventListener("input", updateCity);
+  resetButton.addEventListener("click", ResetCity);
```

Comment:
Looks like we are doing similar logic in a few places (finding an element by `id` and then adding an event handler to it for a specific event). Do you think that this could call for a helper function to make our code a bit more DRY?